### PR TITLE
W-11350149: clean annotations usage and methods

### PR DESCRIPTION
**Pull request by arielmirra**
- W-11350149: make VirtualElement EternalSerialized
- (refactor): rename and add annotation methods
- (refactor): use Eternal trait instead of its mixins
- (refactor): remove duplicated code in annotations

### CODE REVIEW

File: shared/src/main/scala/amf/core/client/scala/model/domain/NamedAmfObject.scala [hunk 1]
```diff
@@ -23,6 +23,8 @@ trait NamedAmfObject extends AmfObject {
 
   def withName(name: String): this.type = withName(ScalarNodeObj(YNode(name)))
 
-  def withSynthesizeName(name: String): this.type = set(nameField, AmfScalar(name), Annotations.synthesized())
+  def withSynthesizedName(name: String): this.type = set(nameField, AmfScalar(name), Annotations.synthesized())
```
Comment (pope1838) [inline shared/src/main/scala/amf/core/client/scala/model/domain/NamedAmfObject.scala]: This is a breaking change

File: shared/src/main/scala/amf/core/internal/parser/domain/Annotations.scala [hunk 2]
```diff
@@ -117,6 +117,14 @@ class Annotations() {
...
   // This should be in amf-shapes BUT PropertyShape is in amf-core
   def isInferredProperty: Boolean = find(classOf[InferredProperty]).isDefined
+
+  def isSynthesized: Boolean = find(classOf[SynthesizedField]).isDefined
+
+  def isVirtual: Boolean = find(classOf[VirtualElement]).isDefined
+
+  def isInferred: Boolean = find(classOf[Inferred]).isDefined
+
+  def clear: Unit = annotations = new ListBuffer()
```
Comment (pope1838) [inline shared/src/main/scala/amf/core/internal/parser/domain/Annotations.scala]: What if we use annotations.clear() instead of create a new list ?

Comment (arielmirra) [reply shared/src/main/scala/amf/core/internal/parser/domain/Annotations.scala]: done

File: shared/src/main/scala/amf/core/client/scala/model/domain/NamedAmfObject.scala (hunk 1 above)
Comment (arielmirra) [reply shared/src/main/scala/amf/core/client/scala/model/domain/NamedAmfObject.scala]: reverted

**Review by pope1838 — APPROVED** [review]

**Pull request merged**